### PR TITLE
feat: add --force flag, e2e test, and doc

### PR DESCRIPTION
**Pull request by iamhopaul123**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by huanjani — APPROVED** [review]
Looks good! This feature will be so helpful for users!

File: site/content/docs/commands/deploy.en.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ This command is used to run either [`copilot svc deploy`](../commands/svc-deploy
 ```bash
   -a, --app string                     Name of the application.
   -e, --env string                     Name of the environment.
+      --force                          Optional. Force a new ECS service deployment using the existing image.
```
Comment (efekarakus) [inline site/content/docs/commands/deploy.en.md]: Do we actually check if it's an ECS service when people run `copilot deploy --force` ?
What's the behavior if they use `--force` on an app runner service? I wonder if for App Runner we can call: https://docs.aws.amazon.com/apprunner/latest/api/API_StartDeployment.html

Comment (iamhopaul123) [reply site/content/docs/commands/deploy.en.md]: <s>Right now if they do `--force` for app runner service then nothing will happen.</s> It throws this error
```
 Failed to force an update for service mysvc from environment test: no ECS service found for mysvc in environment test.
✘ force an update for service mysvc: no ECS service found for mysvc in environment test
```
which should be fixed.

Comment (iamhopaul123) [reply site/content/docs/commands/deploy.en.md]: > I wonder if for App Runner we can call: https://docs.aws.amazon.com/apprunner/latest/api/API_StartDeployment.html

That's a good idea!

Comment (iamhopaul123) [reply site/content/docs/commands/deploy.en.md]: Ok done. Added `--force` for app runner.

File: internal/pkg/cli/svc_deploy.go [hunk 2]
```diff
@@ -570,30 +581,34 @@ func (o *deploySvcOpts) deploySvc(addonsURL string) error {
...
 
+func (o *deploySvcOpts) forceDeploy() error {
+	// Force update the service if --force is set and change set is empty.
+	o.spinner.Start(fmt.Sprintf(fmtForceUpdateSvcStart, color.HighlightUserInput(o.name), color.HighlightUserInput(o.envName)))
+	if err := o.svcUpdater.ForceUpdateService(o.appName, o.envName, o.name); err != nil {
+		errLog := fmt.Sprintf(fmtForceUpdateSvcFailed, color.HighlightUserInput(o.name),
+			color.HighlightUserInput(o.envName), err)
+		var errWaitUpdateTimeout *awsecs.ErrWaitServiceStableTimeout
+		var errWaitOperationFailed *awsapprunner.ErrWaitServiceOperationFailed
+		if errors.As(err, &errWaitUpdateTimeout) || errors.As(err, &errWaitOperationFailed) {
```
Comment (efekarakus) [inline internal/pkg/cli/svc_deploy.go]: Check out this refactor boss: https://play.golang.org/p/VJzxx9PDXG8

What we do here is 1) each error implements the `timeoutError` interface, 2) we check if the error implements the interface in `errors.As`:
```go
var terr timeoutError
if errors.As(err, &terr) {
    ..
}
```

I like this advice from the [Practical Object-Oriented Design book](https://www.poodr.com/) for when we see ourselves checking multiple types:
> When a sending object checks the class of a receiving object to determine what message to send, you have overlooked a duck type.

For Go this means, if we see any statements that have to check multiple data type there is probably an interface that they can implement

**Review by efekarakus — COMMENTED** [review]
Looks great! :shipit:  after the small refactor comment

File: internal/pkg/cli/svc_deploy.go (hunk 2 above)
Comment (iamhopaul123) [reply internal/pkg/cli/svc_deploy.go]: lol

Comment (iamhopaul123) [reply internal/pkg/cli/svc_deploy.go]: Oooh i see what you mean. Good idea!

**Review by efekarakus — APPROVED** [review]

**Pull request merged**